<launch>

  <arg name="gui" default="true" />

  <param name="/use_sim_time" value="true" />

  <node name="rosbag_play"
        pkg="rosbag" type="play"
        args="$(find jsk_pcl_ros)/sample/data/watercan.bag --clock --loop">
  </node>

  <include file="$(find openni2_launch)/launch/openni2.launch">
    <arg name="camera" value="camera"/>
    <arg name="publish_tf" value="false" />
    <arg name="load_driver" value="false"/>
    <arg name="depth_registration" value="true"/>
  </include>

  <arg name="INPUT_CLOUD" value="/euclidean_cluster_indices_decomposer/output00" />

  <node name="detect_graspable_poses_pcabase"
        pkg="jsk_pcl_ros" type="detect_graspable_poses_pcabase.py">
    <remap from="~input" to="$(arg INPUT_CLOUD)" />
    <rosparam subst_value="true">
      direction: z
      hand_size: 0.13
      interval_m: 0.04
    </rosparam>

  </node>

  <group if="$(arg gui)">
    <node name="rviz"
          pkg="rviz" type="rviz"
          args="-d $(find jsk_pcl_ros)/sample/rviz/detect_graspable_poses_pcabase.rviz">
    </node>
  </group>

</launch>
